docs/source/worker.rst

```diff
@@ -218,22 +225,159 @@ usage then the worker will start dumping unused data to disk, even if internal
+    can be tweaked through the ``distributed.worker.memory.recent-to-old-time`` key in
+    the ``~/.config/dask/distributed.yaml`` file. If your tasks typically run for longer
+    than 30 seconds, it's recommended that you increase this setting accordingly.
+
+    By default, :meth:`distributed.Client.rebalance` and
+    :meth:`distributed.scheduler.Scheduler.rebalance` ignore unmanaged recent memory.
+    This behaviour can also be tweaked using the dask config - see the methods'
+    documentation.
+
+spilled
+    managed memory that has been spilled to disk. This is not included in the 'managed'
+    measure above.
+
+The sum of managed + unmanaged + unmanaged recent is equal definition to the process
+memory.
+
+
+Tweaking OS memory allocation
+-----------------------------
+Different OSs use different default policies when it comes to allocating - and, most
+importantly, freeing - memory. Both the Linux and MacOS memory allocators try to avoid
+performing a kernel call every time the application calls ``free()`` by implementing a
+user-space memory management system. Upon ``free()``, memory can remain allocated in
+user space and potentially reusable by the next ``malloc()`` - which in turn won't
+require a system call either. This is generally very desirable for C/C++ applications
+which have no memory allocator of their own, as it can drastically boost performance at
+the cost of a larger memory footprint. CPython however adds its own memory allocator on
+top, which reduces the need for this additional abstraction (with caveats).
+
+Dask.distributed measures the total process memory (also known as RSS) and uses it for
+decision-making in several points; namely:
+
+- the spill, pause, and terminate thresholds (see above)
+- :meth:`distributed.Client.rebalance` and
+  :meth:`distributed.scheduler.Scheduler.rebalance`
+
+In some cases, it is easy to have these heuristics misbehave because there are large
+amounts of process memory that the libc did not release to the OS yet - but Dask does
+not have the means to tell them apart from a memory leak. This phenomenon is the first
+thing you should investigate when you see large amounts of unexplained unmanaged memory
+in the dashboard or in the logs.
+
+Manually trim memory
+~~~~~~~~~~~~~~~~~~~~
+*Linux only*
+
+It is possible to forcefully release allocated but unutilized memory as follows:
+
+.. code-block:: python
+
+    import ctypes
+
+    def trim_memory() -> int:
+        libc = ctypes.CDLL("libc.so.6")
+        return libc.malloc_trim(0)
+
+    client.run(trim_memory)
+
+This should be only used as a one-off debugging experiment.
+
+Automatically trim memory
+~~~~~~~~~~~~~~~~~~~~~~~~~
+*Linux only*
+
+To aggressively and automatically trim the memory in a production environment, you
+should instead set the environment variable ``MALLOC_TRIM_THRESHOLD_`` (note the final
+underscore) to 0 or a low number; see the ``mallopt`` man page for details. Reducing
```

Comment:
Also, out of curiosity, did you ever try calling `mallopt` via ctypes? I know it's a little less safe than the env var, but for people using a managed cluster (pangeo, coiled, DIY, etc.) it may be easier to do a `client.run` than setting pre-startup environment variables.

Reply:
No, because the mallopt documentation stats out loud that you must invoke it before the very first call to malloc - which in CPython is impossible.